### PR TITLE
Validating properties bag for top level ARM resources

### PR DESCRIPTION
**Pull request by dsgouda**
@veronicagg @amarzavery  PTAL
@ravbhatnagar FYI

### CODE REVIEW

**Comment by dsgouda** [comment]
@azuresdkci Test this please

File: src/modeler/AutoRest.Swagger/Model/ServiceDefinition.cs [hunk 1]
```diff
@@ -105,6 +105,7 @@ public ServiceDefinition()
         /// <summary>
         /// Key is the object serviceTypeName and the value is swagger definition.
         /// </summary>
+        [Rule(typeof(ArmResourcePropertiesBag))]
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Model/ServiceDefinition.cs]: should the rule apply only to definitions or could this properties be defined inline?

File: src/core/AutoRest.Core/Properties/Resources.resx [hunk 2]
```diff
@@ -378,4 +378,7 @@ Modelers:
   <data name="TrackedResourcePatchOperationMissing" xml:space="preserve">
     <value>Tracked resource '{0}' must have patch operation that at least supports the update of tags.</value>
   </data>
+  <data name="ArmPropertiesBagValidationMessage" xml:space="preserve">
+    <value>Properties bag [{0}] should not be repeated for ARM top level properties</value>
```
Comment (veronicagg) [inline src/core/AutoRest.Core/Properties/Resources.resx]: Not sure about the clarity of this message. You may want to check with @ravbhatnagar on what he'd like to see.

File: src/modeler/AutoRest.Swagger/Model/ServiceDefinition.cs (hunk 1 above)
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Model/ServiceDefinition.cs]: We discourage inline definitions and there is a rule that flags such violations, so this should do for us.

File: src/core/AutoRest.Core/Properties/Resources.resx (hunk 2 above)
Comment (dsgouda) [inline src/core/AutoRest.Core/Properties/Resources.resx]: Agreed, @ravbhatnagar the message above would look something like "Properties bag [a, b, c, d] should not be repeated for ARM top level properties" where a,b,c,d are the properties found to be violating the rule. Any suggestions here?

Comment (ravbhatnagar) [inline src/core/AutoRest.Core/Properties/Resources.resx]: "Top level property names should not be repeated inside the properties bag. Properties {} conflict with ARM top level properties. Please rename these."

**Pull request merged**